packages/app/src/app/pages/Pro/legacy-pages/WorkspacePlanSelection.tsx

```diff
@@ -1,591 +1,284 @@
-import React from 'react';
+import React, { ButtonHTMLAttributes } from 'react';
 import { format } from 'date-fns';
 import { sortBy } from 'lodash-es';
 import { useLocation } from 'react-router-dom';
-import { Stack, Text, Menu, Icon, Button, Link } from '@codesandbox/components';
-import css from '@styled-system/css';
+import { VisuallyHidden } from 'reakit/VisuallyHidden';
+import { Stack, Text } from '@codesandbox/components';
+import track from '@codesandbox/common/lib/utils/analytics';
 import { useAppState, useActions } from 'app/overmind';
-import { Step, Plan } from 'app/overmind/namespaces/pro/types';
-import { TeamAvatar } from 'app/components/TeamAvatar';
+import { Step } from 'app/overmind/namespaces/pro/types';
+import { SubscriptionType, SubscriptionInterval } from 'app/graphql/types';
+import { useWorkspaceSubscription } from 'app/hooks/useWorkspaceSubscription';
+import { useWorkspaceAuthorization } from 'app/hooks/useWorkspaceAuthorization';
 import {
-  TeamMemberAuthorization,
-  ProSubscription,
-  SubscriptionType,
-  SubscriptionInterval,
-} from 'app/graphql/types';
-import track from '@codesandbox/common/lib/utils/analytics';
-import { plans } from './plans';
+  PERSONAL_FREE_FEATURES,
+  PERSONAL_FEATURES,
+  TEAM_FREE_FEATURES,
+  TEAM_PRO_FEATURES,
+} from 'app/constants';
+import { Switcher } from '../components/Switcher';
+import { SubscriptionCard } from '../components/SubscriptionCard';
+
+const getBillingText = ({
+  quantity,
+  unitPrice,
+  billingInterval,
+}: {
+  quantity: number;
+  unitPrice: number;
+  billingInterval: SubscriptionInterval;
+}) => {
+  const isMonthly = billingInterval === SubscriptionInterval.Monthly;
+  const multiplier = isMonthly ? 1 : 12;
+  const price = quantity * unitPrice * multiplier;
+
+  return `a total of $${price} will be billed each ${
+    isMonthly ? 'month' : 'year'
+  }`;
+};
 
-const prettyPermissions = {
-  ADMIN: 'Admin',
-  WRITE: 'Editor',
-  READ: 'Viewer',
+type CancelButtonProps = {
+  children: string;
+  onClick: ButtonHTMLAttributes<HTMLButtonElement>['onClick'];
 };
 
-export const WorkspacePlanSelection: React.FC<{
-  loading: boolean;
-}> = ({ loading }) => {
+const CancelButton = ({ children, onClick }: CancelButtonProps) => (
+  <Text
+    as="button"
+    variant="danger"
+    onClick={onClick}
+    css={{
+      display: 'block',
+      margin: '0 auto',
+      border: 'none',
+      padding: 0,
+      background: 'transparent',
+      '&:hover': {
+        textDecoration: 'underline',
+      },
+    }}
+  >
+    {children}
+  </Text>
+);
+
+// TODO: Rename
```

Comment:
Missing any action here?

Reply:
Ah yes, I wanted to rename `WorkspacePlanSelection` to something else because it's not really about selecting a plan anymore. Do you think I should rename it in this PR still or will that cause too much unreadable chaos? Since I will also have to change the file name.